### PR TITLE
perf(tag): rendering optimization

### PR DESCRIPTION
**Pull request by SukkaW**
## What does it do?

Escape swig tag inside code block only once.

## How to test

```sh
git clone -b post-tag-render-optimize https://github.com/sukkaw/hexo.git
cd hexo
npm install
npm test
```

## Pull request tasks

- [ ] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

File: lib/extend/tag.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ const { cyan } = require('chalk');
 const { Environment } = require('nunjucks');
 const Promise = require('bluebird');
 const placeholder = '\uFFFC';
-const rPlaceholder = /(?:<|&lt;)!--\uFFFC(\d+)--(?:>|&gt;)/g;
+const rPlaceholder = /<!--\uFFFC(\d+)-->/g;
```
Comment (curbengh) [inline lib/extend/tag.js:8]: I wonder why the previous regex matches `&lt;` `&gt;`? I thought `<>` is only escaped in [code block](https://github.com/hexojs/hexo-util/blob/cd6dcfda34f39cbf992c1e20bb1d4d5570ec7b3c/lib/highlight.js#L104)?

Comment (SukkaW) [reply lib/extend/tag.js:8]: @curbengh IMHO, the regexp here might be copied from `post.js` in which the `<>` will be escaped by post renderer (like marked.js).

But inside tag rendering process, there will be no escape at all.

**Comment by SukkaW** [comment]
@seaoak @curbengh @stevenjoezhang The PR is ready for review now.

**Comment by SukkaW** [comment]
> WFM

@curbengh A lot of test cases has been added during pre-hexo-5.0.0 development and I am sure most of the case has been covered in unit test now.

**Pull request merged**